manifests/systemd.pp

```diff
@@ -0,0 +1,52 @@
+# This class should be considered private
+#
+# This type handles setting up a systemd service and, if applicable, managing
+# the transition from a sysv -> systemd service without leaving zombie services
+# running.
+define jenkins::systemd(
+  $user,
+  $libdir,
+) {
+  if $caller_module_name != $module_name {
+    fail("Use of private class ${name} by ${caller_module_name}")
+  }
+  $service = $name
+
+  include ::systemd
+
+  $sysv_init = "/etc/init.d/${service}"
+
+  file { "${libdir}/${service}-run":
+    content => template("${module_name}/${service}-run.erb"),
+    owner   => $user,
+    mode    => '0700',
+    notify  => Service[$service],
+  }
+
+  transition { "stop ${service} service":
```

Comment:
Case in point, I had to spend several hours debugging the core the in order get it working right with an absent file resource...